src/globals.cpp

```diff
@@ -317,10 +317,14 @@ const sc_string_t *sc_default_target_triple() {
     return get_default_target_triple();
 }
 
-sc_void_raises_t sc_compile_object(const sc_string_t *target_triple,
-    int file_kind, const sc_string_t *path, const sc_scope_t *table, uint64_t flags) {
+sc_void_raises_t sc_compile_object(const sc_string_t *target_triple, int file_kind, const sc_string_t *path, const sc_scope_t *table, uint64_t flags) {
     using namespace scopes;
-    return convert_result(compile_object(target_triple, (CompilerFileKind)file_kind, path, table, flags));
+    return convert_result(compile_object<void>(target_triple, (CompilerFileKind)file_kind, path, table, flags));
+}
+
+sc_string_raises_t sc_compile_object_to_buffer(const sc_string_t *target_triple, int file_kind, const sc_string_t *module_name, const sc_scope_t *table, uint64_t flags) {
```

Comment:
Tested a few times, I was reading through documentation when I was writing this but the LLVM doc is kinda rough. Regardless of input the output binary shows no difference on the ASM, not sure about the file but I imagine it is just stored internally for the module short name.

Reply:
I'm pretty sure the only thing the module_name needs to be is unique. There should be a function in the compiler that generates unique IDs that could be used, I think, instead of exposing it to scopes users and requiring them to ensure the module name is unique.